src/Stratis.Bitcoin.Tests/Controllers/NodeControllerTest.cs

```diff
@@ -61,11 +63,16 @@ public NodeControllerTest()
         }
 
         [Fact]
-        public void Stop_WithFullNode_DisposesFullNodeAsync()
+        public void Stop_WithFullNode_DisposesFullNode()
         {
+            var isDisposed = false;
+            this.fullNode.Setup(f => f.Dispose()).Callback(() => isDisposed = true);
+
             IActionResult result = this.controller.Shutdown(0);
 
-            this.fullNode.Verify(f => f.Dispose());
+            result.Should().BeOfType<OkResult>();
+            Thread.Sleep(100);
```

Comment:
Ideally I would have brought the waitloop from integration tests or raised a public event to test, but I thought this would be deemed out of scope. Hopefully this is not too bad.